sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,845 @@
+`M` rows by `K` columns, and the `B` matrix is `K` rows by `N` columns.
+
+==== Compile-Time Query
+This returns `constexpr` values to use in `joint_matrix` template
+arguments but depends on an enumeration of the matrix hardware (See
+`sycl::ext::oneapi::experimental::architecture`) in the
+link:../sycl_ext_oneapi_device_architecture.asciidoc[sycl_ext_oneapi_device_architecture]
+extension that can be tested.
+The compile-time query interface proposed here consists of two
+functionalities:
+
+- Validation: at compile time, the validation functionality informs
+  the user whether a specific combination is valid or not. This takes
+  place when the user specifies all template parameters.
+
+- Default values: this provides a default shape if the user does not
+  provide a specific combination. In this case, aliases to the
+  `joint_matrix` type can be used, namely
+  `joint_matrix_a/b/c/d` where no additional argument is needed. This
+  form happens when the user specifies all template parameters except
+  the sizes of the matrices M, N, and K.
+
+The table below provides a description for each of the member
+variables in `matrix_params` class and the forms in which  they are
+defined.
+
+[frame="none",options="header"]
+|======================
+| Member/type alias in `matrix_params` | Description
+|`M`|when no sizes are provided by the user, indicates the suggested
+default size for M; usually this corresponds to the maximum size the
+implementation supports. In validation mode, where the user does
+provide sizes, this is the same value M that the user provides if M is
+supported by the implementation
+|`N`|when no sizes are provided by the user, indicates the suggested
+default size for N; usually this corresponds to the maximum size the
+implementation supports. In validation mode, where the user does
+provide sizes, this is the same value N that the user provides if N is
+supported by the implementation
+|`K`| when no sizes are provided by the user, indicates the suggested
+default size for K; usually this corresponds to the maximum size the
+implementation supports. In validation mode, where the user does
+provide sizes, this is the same value K that the user provides if K is
+supported by the implementation
+|`template <typename Group, layout Layout> +
+using joint_matrix_a;`| type alias for `joint_matrix` for matrix A
+|`template <typename Group, layout Layout> +
+using joint_matrix_b;`| type alias for `joint_matrix` for matrix B
+|`template <typename Group> +
+using joint_matrix_c;`| type alias for `joint_matrix` for the input matrix accumulator
+|`template <typename Group> +
+using joint_matrix_d;`| type alias for `joint_matrix` for the output matrix accumulator
+|======================
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template<architecture Arch, typename Ta=void, typename Tb=void,
+typename Tc=void, typename Td=Tc, int sM=0, int sN=0, int sK=0> 
+struct matrix_params;
```

Comment:
```suggestion
template<architecture Arch, typename Ta, typename Tb,
typename Tc, typename Td=Tc, size_t sM=0, size_t sN=0, size_t sK=0> 
struct matrix_params;
```

Your latest commit created this new primary template definition.   This is OK, but it wasn't necessary.  In the previous version, the definition of the "validation form" also served as the primary template definition.

If you want to do it this, way, I think you should have the changes I show above:

* As before, there is no reason to have default values for `Ta`, `Tb` and `Tc`.  Aren't these template parameters required in both "validation form" and "default values form"?
* The types of `sM`, etc. must be `size_t` (not `int`) to match the specializations below.

Reply:
the validation form should be also a specialization as we would specialize on the Arch type there: architecture::intel_gpu_pvc. I am correcting that.